src/vsr/replica.zig

```diff
@@ -1154,7 +1175,9 @@ pub fn ReplicaType(
             if (self.heartbeat_timestamp < message.header.timestamp) {
                 self.heartbeat_timestamp = message.header.timestamp;
                 self.normal_heartbeat_timeout.reset();
-                self.start_view_change_from_all_replicas.unset(self.replica);
+                if (!self.standby()) {
+                    self.start_view_change_from_all_replicas.unset(self.replica);
```

Comment:
This one is a bit ugly, this is where "not doing something as late as possible breaks": quorum counters a bitsets of `replicas_max` size, so, although they are logically unset for all higher indexes and that would be correct, in code we hit panics if `self.replica` points to a standby. 

Alternatives: 

* guard bitset ops with `if (!self.standby())` (what I ended up doing)
* extend `QuorumCounter`'s size to `nodes_max`, rather than `replicas_max` (don't like this, as we do want to assert that we never try to add standby to the quorum)
* for `QuorumCounter`, wrap the bitset into a type which knows replicas_count and standby_count, which asserts that we never try to set standby, and we never use any inxeses larger than `node_max`. I like this in principle, but I don't like adding extra abstractions.

Reply:
> extend `QuorumCounter`'s size to `nodes_max`, rather than `replicas_max` (don't like this, as we do want to assert that we never try to add standby to the quorum)

This is ugly, but could live in each quorum reset function:

```zig
assert((self.start_view_change_from_all_replicas.iterator(.{ .direction = .reverse }).next() orelse 0) < self.replica_count);
```

(Though it would only discover adding standbys after the fact.)